framework/src/meshgenerators/PatternedMeshGenerator.C

```diff
@@ -88,35 +116,68 @@ PatternedMeshGenerator::generate()
                  type(),
                  " only works with inputs that are replicated.\n\n",
                  "Try running without distributed mesh.");
-    _meshes.push_back(std::move(mesh));
-  }
+    _meshes[i] = dynamic_pointer_cast<ReplicatedMesh>(mesh);
 
-  // Data structure that holds each row
-  _row_meshes.resize(_pattern.size());
+    const auto ids = MooseMeshUtils::getBoundaryIDs(*_meshes[i], boundary_names, false);
+    mooseAssert(ids.size() == boundary_names.size(),
+                "Unexpected number of ids returned for MooseMeshUtils::getBoundaryIDs");
 
-  // Getting the boundaries provided by the user
-  std::vector<BoundaryName> boundary_names = {getParam<BoundaryName>("left_boundary"),
-                                              getParam<BoundaryName>("right_boundary"),
-                                              getParam<BoundaryName>("top_boundary"),
-                                              getParam<BoundaryName>("bottom_boundary")};
+    // Keep track of used IDs so we can later find IDs that are unused across all meshes
+    used_boundary_ids.insert(ids.begin(), ids.end());
```

Comment:
this isnt a full list of used ids. It's only ids used for the the names op top/left/...

In ids, you need to get ALL ids in those meshes to find ones that are never used. 

Otherwise when you change ids you ll hit existing ids that are not part of top/left/...

Reply:
but then your use of ids below is different. 

I reckon you should create a new all_ids set here, then use all_ids for used_boundary_ids